docs/dev/test-indexes.md

```diff
@@ -0,0 +1,119 @@
+
+The bundles are useful independently of any catalog as they represent a specific point in time for an operator publisher. We can simulate arbitrary operator publishes by generating arbitrary bundles, with upgrade graph annotations (`replaces`, `skips`, and the more complicated `skipRanges`). So, define the bundles that represent the operator publishing actions you'd like to simulate. Note that the examples given above for the `foo` package are actually bundles currently written in the test data. When we're filling out these bundles, you should follow the lead of the `foo` operator and ensure that all container image references are marked as `REGISTRY_CATALOGNAMESPACE:<operator>-<version>`, as seen for `foo-v.0.1.0` [here](test/operator/bundles/foo/foo-bundle-v0.1.0/manifests/foo.csv.yaml#L8)
+
+Organizing those bundles into snapshot-in-time catalogs is done via definition in [the publish script](test/operator/publish_images.py). The catalogs are organized into Python dictionaries with the package name as the key and the bundle versions in a list as the value of those keys. Catalog names are then defined in the [CATALOGS global variable](test/operator/publish_images.py#L36) as dictionary keys, with the dictionary of packages as the value. Channels in the catalogs are built from the bundle metadata, and bundles are organized into those channels as the catalog is built.
+
+If you need to make a change to one of the test catalogs to simulate a different point in time across updates, or to publish new bundles with different data/metadata into one of the existing catalogs, these are the places you would update them.
+
+Execution of the script with no options will gather the specified bundles into catalogs, generate bundle images, generate fake "operator" images that only echo their originally published name and then sleep, and render and generate the catalog images. These are all pushed to the specified registry as this process goes. `CatalogSource` resources are generated per catalog and sent to STDOUT for application to a cluster.
+
+Environment variables or command line flags can override the registry, registry repository (or catalog namespace), container runtime, catalog selection, file-based catalog architecture, and apply a catalog to a cluster automatically. The command-line flags are best documented by the help screen for the script, visible at `publish_images.py -h`. Images should only be published to the default registry and catalog namespace from `main` to ensure that the main tests are drawing from known catalogs. If you want to publish alternative catalogs/bundles with local changes to test something before merging to main, the easiest way is to use either the `REGISTRY` or `CATALOG_NAMESPACE` variable to override the defaults. For example:
+
+```sh
+REGISTRY=registry.jharmison.com ./publish_images.py
```

Comment:
Are we going to change the reference here to `quay.io`?

Reply:
That's the default, this is to demonstrate how to override the registry references in every bundle and push to an alternative registry.